src/quo2/components/wallet/wallet_activity/view.cljs

```diff
@@ -0,0 +1,169 @@
+(ns quo2.components.wallet.wallet-activity.view
+  (:require [quo2.theme :as quo.theme]
+            [quo2.components.wallet.wallet-activity.style :as style]
+            [react-native.core :as rn]
+            [quo2.components.markdown.text :as text]
+            [quo2.components.icon :as icon]
+            [quo2.components.tags.context-tag.view :as context-tag]
+            [react-native.hole-view :as hole-view]
+            [utils.i18n :as i18n]
+            [reagent.core :as reagent]))
+
+(def transaction-translation
+  {:receive [i18n/label :t/receive]
+   :send    [i18n/label :t/send]
+   :swap    [i18n/label :t/swap]
+   :bridge  [i18n/label :t/bridge]
+   :buy     [i18n/label :t/buy]
+   :destroy [i18n/label :t/destroy]
+   :mint    [i18n/label :t/mint]})
+
+(def transaction-icon
+  {:receive :i/receive
+   :send    :i/send
+   :swap    :i/swap
+   :bridge  :i/bridge
+   :buy     :i/buy
+   :destroy :i/destroy
+   :mint    :i/mint})
+
+(def status-icon
+  {:pending   :i/pending-state
+   :confirmed :i/positive-state
+   :finalised :i/diamond-blue
+   :failed    :i/negative-state})
+
+(defn transaction-header
+  [{:keys [transaction
+           timestamp
+           counter
+           theme
+           blur?]
+    :or   {transaction :receive
+           counter     1}}]
+
+  [rn/view
+   {:style style/transaction-header-container}
+   [text/text
+    {:weight :semi-bold
+     :size   :paragraph-1
+     :style  (style/transaction-header theme)}
+    (transaction transaction-translation)]
+   (when (> counter 1)
+     [rn/view (style/transaction-counter-container theme blur?)
+      [text/text
+       {:weight :medium
+        :size   :label
+        :style  (style/transaction-counter theme)}
+       (str "x" counter)]])
```

Comment:
IMO is better to use localized strings here, because of future RTL languages support